File: CS/Base64Converter.ascx.cs
using System;
using DevExpress.Web.ASPxUploadControl;
using System.Web.UI;
using System.IO;
using System.Drawing;

public partial class Base64Converter : System.Web.UI.UserControl {
    const string UploadDirectory = "~/UploadImages/";
    const string ThumbnailFileName = "ThumbnailImage.jpg";

    protected void uplImage_FileUploadComplete(object sender, FileUploadCompleteEventArgs e) {
        e.CallbackData = SavePostedFile(e.UploadedFile);
    }

    string SavePostedFile(UploadedFile uploadedFile) {
        string ret = "";
        string base64String = "";

        if (!uploadedFile.IsValid)
            return string.Empty;

        string fileName = Path.Combine(MapPath(UploadDirectory), ThumbnailFileName);
        using (Image original = Image.FromStream(uploadedFile.FileContent)) {
            base64String = PhotoUtils.ConvertImageToBase64(original);
            using (Image thumbnail = PhotoUtils.Inscribe(original, 100)) {
                PhotoUtils.SaveToJpeg(thumbnail, fileName);
            }
        }


        ret = string.Format("{0}|{1}", ThumbnailFileName, base64String);

        //return ThumbnailFileName;
        return ret;
    }
}

File: VB/Base64Converter.ascx.vb
Imports System
Imports DevExpress.Web
Imports System.Web.UI
Imports System.IO
Imports System.Drawing

Partial Public Class Base64Converter
    Inherits System.Web.UI.UserControl

    Private Const UploadDirectory As String = "~/UploadImages/"
    Private Const ThumbnailFileName As String = "ThumbnailImage.jpg"

    Protected Sub uplImage_FileUploadComplete(ByVal sender As Object, ByVal e As FileUploadCompleteEventArgs)


        e.CallbackData = SavePostedFile(e.UploadedFile)
    End Sub

    Private Function SavePostedFile(ByVal uploadedFile As UploadedFile) As String
        Dim ret As String = ""
        Dim base64String As String = ""

        If Not uploadedFile.IsValid Then
            Return String.Empty
        End If

        Dim fileName As String = Path.Combine(MapPath(UploadDirectory), ThumbnailFileName)
        Using original As Image = Image.FromStream(uploadedFile.FileContent)
            base64String = PhotoUtils.ConvertImageToBase64(original)
            Using thumbnail As Image = PhotoUtils.Inscribe(original, 100)
                PhotoUtils.SaveToJpeg(thumbnail, fileName)
            End Using
        End Using


        ret = String.Format("{0}|{1}", ThumbnailFileName, base64String)

        'return ThumbnailFileName;
        Return ret
    End Function
End Class